src/uu/cp/src/cp.rs

```diff
@@ -1883,6 +1883,7 @@ fn copy_file(
     Ok(())
 }
 
+#[allow(clippy::unnecessary_cast)]
```

Comment:
do you know many occurrences of this happen in this function? If not too many, could you please move that into the function itself

Reply:
Actually, looking at the function again, I see there's already a seperate case with an `as u32` for FreeBSD / macOS / Android which are using `u16`. So Redox could be moved to that case. Though just using a cast with this `#[allow]` is simpler.

I wonder if it would be better to just not use these constants, and just use octal notation here? Their values should be standardized, and I think people may be more familiar with the octal values for Unix-style permissions than with the  constant names.